### ISSUE TITLE
License conflict and indemnity. (Copyleft compliance)

### ISSUE THREAD

**Issue by StormTide** [BCDevExchange/rapid-adoption#3]
As a developer the Terms https://github.com/BCDevExchange/rapid-adoption/wiki/Pay-For-Pull-Terms place additional contractual obligations above and beyond an open license I am comfortable using. 

Specifically when modifying an open source project such as Wordpress (per Sage/Wordpress-related opportunities https://github.com/bcgov/citizen-engagement-web-toolkit/issues/5 ) the Terms could be read to conflict with the license rights of Wordpress (Copyleft via GPLv2) https://wordpress.org/about/license/ and the Indemnity clause would hold me, the developer, responsible for breaching those license terms by licensing my contributions to the program under the Apache 2.0 license as required by the pull request. This despite the inducement to use Apache 2.0 found here:  https://github.com/bcgov/citizen-engagement-web-toolkit/issues/5 ... 

Proposed fix: Waive the indemnity, review the terms with the community, and adopt the compliant licensing for the pull requests as required by the upstream project in which the opportunity exists.

**Comment by DavidR8** [comment]
Hi Kevin, appreciate your comments on this.
The developers of the Sage code provided with this issue put the code under a non-GPL license that allows for sub-licensing. Use of the Apache 2.0 license for any pull requests doesn’t create a breach of GPL terms. 
https://github.com/bcgov/citizen-engagement-web-toolkit/blob/master/code/bcgov-comment-theme/LICENSE.md

**Comment by StormTide** [comment]
Au contraire. Wordpress themes are derivative works of Wordpress, a GPL licensed open source program. In fact I linked to their license in the comment above.  

I've been around F/LOSS licensing for close to two decades now so this is 'kind of my wheelhouse'.

Here's Wordpress's own legal opinion on the GPL's applicability to Themes.. https://wordpress.org/news/2009/07/themes-are-gpl-too/ There's about 100 trackbacks you can read that summarize the controversy, and volumes have been penned on this topic. Needless to say its settled in the community, GPL's Copyleft infects the PHP code of Wordpress themes. This PHP code is what is in question with this paid pull opportunity.

I cannot enforce Wordpress's license rights per Sage, and per Sage's license they disclaim downstream rights of noninfringement, so any downstream user mistakenly tripping over Wordpress's GPL license is on their own.

What got missed here was an evaluation on whether Sage had the right to license a non-GPL Wordpress Theme. Cleverly the Pay-For-Pull indemnity terms put that mistake firmly on the shoulders of the downstream participating developer. 

Were I to complete this Pay For Pull opportunity, I would violate the rights of a third party (Wordpress in creating a derivative work, and per a third party legal opinion), run afoul of the GPL and find myself liable. The custom license per Sage may allow sub-licensing, but that is not sufficient to find itself bidirectionally compatible with the FSF GPL license (which requires that downstream software, even those combined with a GPL-compatible work, also carry the GPL license), or to save me from the Pay-For-Pull Terms' indemnity clause. That is to say, I cannot disclaim non-infringement as the Sage developers did, as the Pay-For-Pull Terms create additional restrictions.

All that is to say, one cannot derive a GPL (copyleft work) into a MIT, Apache2 or custom-licensed work. One may include GPL-compatible works into a GPL-covered work, but the resulting work must be GPL. If theres any confusion about this, I'd be happy to close the loop with the Wordpress developers so that issues of copyleft licensing can be explained more effectively than perhaps I am able to.

TL;DR You stepped on the minefield which is GPL licensing in Wordpress, and might now have a better understanding of why developers worry so much about licensing, indemnities, warranties and so forth.

**Comment by StormTide** [comment]
I thought I should add some solutions here; for fear of sounding like I'm complaining without offering a solution. 

A couple possible resolutions to this issue would look like.

Scenario 1. (Attempt compliance, aka Maintaining the induce-and-indemnify policy stack)

1 - Fork Sage to GPL. 

This is compliant per the Sage license, and brings compliance with the upstream GPL-licensed Wordpress. The Wordpress license is complied with re derivative works, and the sage license is complied with per re-licensing. Downstream use will be GPL, as is required by Wordpress's copyleft. Contribution back to the custom-licensed Sage branch would not be possible.

2 -Update the Pay for Pull opportunity to require contributions under the GPL license /for these GPL-derived opportunities/

3 - Leave the Terms in place and hope a developer is willing to risk that there are other infringing uses in Sage and give the province an indemnity.

Scenario 2. (Insure against non-compliance. Drop induce-and-indemnify)

1 - Drop the Pay-For-Pull terms.

The Terms place additional restrictions and warranties above and beyond the license rights generally associated with a F/LOSS project. Per Sage, noninfringement liability to downstream users is specifically disclaimed (upstream liability to Wordpress for down-licensing cannot be escaped but the network effect can be limited)

2 - Require contributions under the same license as the parent project. 

Using Sage's license for further contributions would allow the developer of the pay-for-pull improvements to disclaim downstream liability for noninfringement -- leaving the compliance matter between the Wordpress Authors and the Developer, rather than a nebulous number of downstream users and the province. 

3 - Find an appropriate Insurance product for the province and potentially for the developer for errors and omissions wrt Copyright Infringement, inducement and the like. The Wordpress Authors may or may not choose to pursue the non-compliance and the insurer would be the liable party in the event this risk materialized. For most OSS projects, enforcing compliance is outside the budget but held as a sword-of-damocles to deal with any serious issues by bad actors. GPL compliance issues are almost never resolved in legal process, but they have been in some key cases Wallace, TiVO and the like. Finding an insurer willing to take this risk for a reasonable fee may be possible. (Read: This is similar to how Uber handles insuring drivers participating in their grey-market programs)

**Comment by DavidR8** [comment]
Hey @StormTide, thanks much for your reply.
We are going to change the license to the MIT/X11 license for the Pay for Pull issues under https://github.com/bcgov/citizen-engagement-web-toolkit. This will keep everything inline with the licensing approach taken by the Sage creators. 
FYI @coakenfold

**Comment by coakenfold** [comment]
I created a plugin as part of my Pay-For-Pull solution and didn't (knowingly) alter Sage/Wordpress. 

Does submitting a plugin sidestep all of these issues?

I haven't considered these licensing issues before so please excuse any ignorance on my part.

**Comment by DavidR8** [comment]
Hey Chad, no worries.
We will ensure that all licensing aspects are in order.
Thanks again for your interest and pull request!

David

On Feb 4, 2016, at 11:45 AM, Chad Oakenfold <notifications@github.com<mailto:notifications@github.com>> wrote:

I created a plugin as part of my Pay-For-Pull solution and didn't (knowingly) alter Sage/Wordpress.

Does submitting a plugin sidestep all of these issues?

I haven't considered these licensing issues before so please excuse any ignorance on my part.

##

**Comment by StormTide** [comment]
@DavidR8 sorry, not sure how changing from apache2 to mit/x11 changes the calculus here? Are you also dropping the indemnity's additional terms?

**Comment by StormTide** [comment]
Maybe this video helps clear things up wrt Wordpress's GPL license? https://videopress.com/v/ABaVkvrA

**Comment by StormTide** [comment]
This might also be helpful for those who don't really understand what Copyleft (vs copyright) means and what viral licensing is. http://www.gnu.org/licenses/copyleft.en.html

The province might hope this to be about Sage and Pay-For-Pull and contributing like-to-like, but theres a set of third parties known as the "Wordpress Authors" who have license rights upstream of Sage, and that "pollute" the downstream code environment with GPL's copyleft terms. This is a good thing in terms of keeping the Wordpress ecosystem GPL licensed and according those fundamental copyleft freedoms, but its in conflict with an approach that grants the additional rights to close the source (per MIT/X11 or Apache2). Sage (to the extent they create a derivative work of Wordpress) doesn't have the upstream rights to strip off Wordpress's GPL terms and relicense under a more open (non-copyleft) custom license -- so this is the first part of the compliance problem. The province relied upon the new more open (non-copyleft) terms per Sage, but without doing a necessary upstream rights check to determine if Sage itself was compliant with upstream licensing requirement from Wordpress. The Sage license also told the Province that it wasn't going to be liable to the Province for infringement. (They could still be liable to the Wordpress Authors, but thats between the Wordpress Authors and Sage)...

That brings us full circle to where the Pay-For-Pull developer should care. If they were just contributing to Sage under Sage's license, they'd still have an open liability to the Wordpress Authors insofar as there was a GPL license violation, but could disclaim downstream responsibility to people who rely on Sage's code -- but then, The Pay-For-Pull terms, extend the Sage license and require indemnity, warranties of owning the copyright, and so forth. That places the Pay-For-Pull developer not only liable to the Wordpress Authors, but also to the Province, and any downstream users from that point on. Worse the indemnity means that should the Province publish the Pay-For-Pull code, and find themselves liable to the Wordpress Authors, that the Pay-For-Pull developer will be responsible for that liability. In essence, the province covers its liabilities while inducing the developer to take on massive liabilities and license code in a way that is known to conflict with the Wordpress Authors GPL license upstream. 

All that is to say, I dont think this opportunity is even completable as spec'd out today in a way that would not find the developer liable to the Wordpress Authors -- and with the additional indemnities and warranties found in the Pay-For-Pull Terms the developer takes all the risk for copyright infringement, even though such infringement is almost certainly core to the task at hand, and the provinces request for license under non-gpl terms could be considered an inducement to infringement.

**Comment by DavidR8** [comment]
Hi Kevin, for my own clarification I need to understand something.

Near the end of the article you provided: https://wordpress.org/news/2009/07/themes-are-gpl-too/ 
it states "...we will only promote and host things on WordPress.org that are 100% GPL or compatible."
The MIT license or X11 as it's more formally known is GPL compatible according to the Free Software Foundation.

So I don't understand how Sage is offside by putting their theme under the MIT license.
It seems to me they are simply taking the "GPL compatible" route that WordPress has publicly said is acceptable.
